qiskit/circuit/quantumcircuit.py

```diff
@@ -1941,6 +1961,29 @@ def _bind_parameter(self, parameter, value):
                 parameter in self.global_phase.parameters):
             self.global_phase = self.global_phase.bind({parameter: value})
 
+    def _bind_all_parameters(self, param_dict):
+        """Assigns parameter values to matching instructions in-place."""
+        for param in self._parameter_table:
+            for (instr, param_index) in self._parameter_table[param]:
+                if not isinstance(instr.params[param_index], ParameterExpression):
+                    continue
```

Comment:
Would these be cases where the `ParameterTable` is invalid (referencing a `Parameter` where one isn't used in the circuit)? Would it be better to `raise` in these cases?